-Dlocal -DskipTests still fails after #8533 — app test sources import classes from modules the local tier excludes

Registry version: main (`05667bf`)
OS: Windows 11, Maven wrapper, Temurin 21.0.11

This is a follow-up to #8522 / #8533 — thanks for the quick fix there! The `local-excluded-test-deps` profile fixes dependency *resolution* under `-Dlocal`, so the build gets further now, but the README getting-started sequence still fails — at test *compilation* this time. (This is the separate problem I mentioned at the end of #8966.)

**Steps to reproduce** (fresh clone of main, JDK 21):

```
./mvnw clean install -Dlocal -DskipTests
```

**Actual:** the `app` module fails at `default-testCompile`:

```
[ERROR] COMPILATION ERROR :
[ERROR] .../app/src/test/java/io/apicurio/registry/AbstractClientFacadeTestBase.java:[6,44] package io.apicurio.registry.resolver.client does not exist
[ERROR] .../app/src/test/java/io/apicurio/registry/auth/MojoAuthTest.java:[5,34] package io.apicurio.registry.maven does not exist
[ERROR] .../app/src/test/java/io/apicurio/registry/customTypes/WebhookArtifactTypesTest.java:[3,44] package io.apicurio.utils.test.raml.microsvc does not exist
[ERROR] .../app/src/test/java/io/apicurio/registry/noprofile/JsonSerdeTest.java:[5,41] package io.apicurio.registry.serde.config does not exist
... (many more — 21 test files affected in total)
```

**Why:** `-DskipTests` skips test *execution* but still compiles `app/src/test`. Under `-Dlocal`, the `local-excluded-test-deps` profile in `app/pom.xml` (active on `!local`) correctly removes the six test-scoped artifacts so resolution succeeds — but the test sources that need them still get compiled: 11 files import `io.apicurio.registry.serde.*`, 9 import `io.apicurio.registry.maven.*`, 1 imports `io.apicurio.utils.test.raml.microsvc.*`, plus `io.apicurio.registry.resolver.client` imports that arrive transitively through those artifacts. So after #8533, resolution passes but compilation doesn't.

**The README getting-started (lines 62–66) hits this from both directions:**

```
./mvnw clean install -Dlocal -DskipTests
cd app/
../mvnw quarkus:dev
```

Step 1 with `-Dlocal` never builds or installs the serdes / maven-plugin / raml artifacts into `~/.m2`, and step 3 runs `quarkus:dev` *without* `-Dlocal` — so the `!local` profile activates and demands those six artifacts, which fails on a machine that has only ever done `-Dlocal` builds. And adding `-Dlocal` to the `quarkus:dev` step doesn't help either: dev mode compiles `app/src/test` too, so it fails with the same compilation errors before the server starts.

**Workaround (verified):** use `-Dmaven.test.skip=true` (skips test *compilation*, not just execution) on both steps:

```
./mvnw clean install -Dlocal -Dmaven.test.skip=true
cd app/
../mvnw quarkus:dev -Dlocal -Dmaven.test.skip=true
```

With that, dev mode starts normally and the API responds.

**Possible fixes** — happy to send a PR for whichever direction you prefer:

1. Smallest: update the README quick-start and build-tier docs to use `-Dlocal -Dmaven.test.skip=true` on both steps.
2. More robust: in `app/pom.xml`, add a `local`-activated profile configuring maven-compiler-plugin `<testExcludes>` for the affected test sources, so `-Dlocal -DskipTests` and `quarkus:dev -Dlocal` work as-is.
3. Longer-term: move the serde/mojo/raml tests out of `app` into the modules that own those features, so `app` doesn't test-depend on modules the local tier excludes.

--- a/.claude/commands/run-tests.md
+++ b/.claude/commands/run-tests.md
@@ -29,7 +29,7 @@ $ARGUMENTS — Optional: module name, test class, or "all"
 
 4. **If storage layer was changed**, suggest integration tests:
    ```bash
-   ./mvnw verify -pl integration-tests -Plocal-tests
+   ./mvnw verify -Pintegration-tests -pl integration-tests -am
    ```
 
 5. **If UI was changed**:

--- a/CLAUDE.md
+++ b/CLAUDE.md
@@ -10,7 +10,7 @@ Open-source API and Schema Registry. Apache 2.0 license, DCO sign-off required o
 ./mvnw quarkus:dev                       # Dev mode (run from app/)
 ./mvnw test -pl <module>                 # Run tests for a specific module
 ./mvnw checkstyle:check -pl <module>     # Run checkstyle for a module
-./mvnw verify -pl integration-tests -Plocal-tests  # Integration tests
+./mvnw verify -Pintegration-tests -pl integration-tests -am  # Integration tests
 ```
 
 UI (separate build system):

--- a/CONTRIBUTING.md
+++ b/CONTRIBUTING.md
@@ -80,11 +80,38 @@ Because we are all humans, and to ensure Apicurio Registry is stable for everyon
 Don't forget to include tests in your pull requests.
 Also don't forget the documentation (reference documentation, javadoc...).
 
-Be sure to test your pull request using all storage variants:
+Be sure to test your pull request against the storage variants your change affects.
 
-1. SQL storage (using the `-Psql` profile)
-2. SQL Server storage (using the `-Pmssql` profile)
-3. KafkaSQL storage (using the `-Pkafkasql` profile)
+Since Apicurio Registry 3.0 a single build supports every storage variant, so the
+variant is selected at runtime rather than by a Maven profile
+(see [DEVELOPING.md](DEVELOPING.md#build-configuration)):
+
+| Storage variant                | Selected with                                                            |
+|--------------------------------|--------------------------------------------------------------------------|
+| SQL                            | `-Dapicurio.storage.kind=sql` (default)                                   |
+| KafkaSQL                       | `-Dapicurio.storage.kind=kafkasql`                                        |
+| GitOps (experimental)          | `-Dapicurio.storage.kind=gitops`                                          |
+| Kubernetes ConfigMap (experimental) | `-Dapicurio.storage.kind=kubernetesops`                              |
+
+For the SQL variant, the database flavor is chosen separately with
+`-Dapicurio.storage.sql.kind`, which accepts `h2` (default), `postgresql`, `mssql`,
+and `mysql`.
+
+Storage-specific unit tests live under the matching packages and can be run directly:
+
+```bash
+./mvnw test -pl app -Dtest='io.apicurio.registry.storage.impl.kafkasql.**'
+```
+
+Integration tests are opt-in and are documented in the
+[integration tests module](integration-tests/):
+
+```bash
+./mvnw verify -Pintegration-tests -pl integration-tests -am
+```
+
+CI runs the full storage matrix on every pull request, so running every variant
+locally is not required.
 
 ### Customizing Registry supported ArtifactTypes
 

--- a/DEVELOPING.md
+++ b/DEVELOPING.md
@@ -55,13 +55,23 @@ The project uses a three-tier build system to allow developers to build only wha
 
 ```bash
 # Local: core server only, skip non-essential plugins (~3 min)
-./mvnw clean install -Dlocal -DskipTests
+./mvnw clean install -Dlocal -Dmaven.test.skip=true
 
 # Default: normal development
 ./mvnw clean install -DskipTests
 
 # Full: everything
 ./mvnw clean install -Dfull -DskipTests
+```
+
+**Note:** the *local* tier requires `-Dmaven.test.skip=true` rather than `-DskipTests`.
+The local tier excludes the serde and Maven plugin modules, but the `app` module's test
+sources import packages from them, so `-DskipTests` (which still compiles test sources)
+fails at test compilation. `-Dmaven.test.skip=true` skips test compilation entirely.
+The same applies to dev mode, which also compiles test sources:
+
+```bash
+cd app && ../mvnw quarkus:dev -Dlocal -Dmaven.test.skip=true
 ```
 
 Integration tests and examples are always opt-in via their own profiles:
@@ -72,7 +82,8 @@ Integration tests and examples are always opt-in via their own profiles:
 | Property              | Purpose                                                                                  |
 |-----------------------|------------------------------------------------------------------------------------------|
 | `-Pprod`              | Enables Quarkus *prod* configuration profile (higher logging level, production defaults) |
-| `-DskipTests`         | Skip all tests                                                                           |
+| `-DskipTests`         | Skip running tests (test sources are still compiled)                                     |
+| `-Dmaven.test.skip=true` | Skip compiling and running tests (required with `-Dlocal`)                            |
 | `-DcliSkipNative`     | Skip CLI native image compilation (no executable is produced, but tests can still run)   |
 | `-DskipOperatorTests` | Skip operator tests (default: `true`, requires a running cluster)                        |
 
@@ -132,7 +143,7 @@ are in a separate module and need to be explicitly enabled:
 
 ```bash
 # Run default integration tests (smoke + serdes + acceptance)
-./mvnw verify -Pintegration-tests -Plocal-tests -pl integration-tests -am
+./mvnw verify -Pintegration-tests -pl integration-tests -am
 ```
 
 See the [integration tests module](integration-tests/) for test groups, deployment

--- a/README.md
+++ b/README.md
@@ -14,10 +14,13 @@ Build the project and run the registry with the in-memory storage variant:
 **Build requirement:** JDK 21 or newer is required to build the project (the build tooling, e.g. Checkstyle, needs a Java 21+ runtime). The produced artifacts still target Java 17.
 
  ```
- ./mvnw clean install -Dlocal -DskipTests
+ ./mvnw clean install -Dlocal -Dmaven.test.skip=true
  cd app/
- ../mvnw quarkus:dev
+ ../mvnw quarkus:dev -Dlocal -Dmaven.test.skip=true
  ```
+
+(The `-Dlocal` build tier requires `-Dmaven.test.skip=true` — see
+[DEVELOPING.md](DEVELOPING.md#build-tiers) for details and other build options.)
 
 This should result in Quarkus and the in-memory registry starting up, with the REST APIs available on localhost port 8080:
 

